conformance/ion_encoding/module/macro/cardinality/invoke_cardinality_ee.ion

```diff
@@ -0,0 +1,151 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Tests covering cardinality and grouping in E-expression macro invocations.
+
+
+(ion_1_1 "! parameters"
+  (mactab
+    (macro m (v!) v))
+  (each (toplevel ('#$:m'))
+        (signals "Too few arguments"))
+  (each (toplevel ('#$:m' ('#$:')))
+        (toplevel ('#$:m' ('#$:' 1)))
+        (toplevel ('#$:m' ('#$:' 1) 2))             // Hidden: too many args
+        (toplevel ('#$:m' ('#$:' 1 2)))
+        (signals "Unexpected argument group"))
+  (each (toplevel ('#$:m' ('#$:void')))
+        (signals "No value provided for parameter"))
+  (each (toplevel ('#$:m' ('#$:values' 1 2)))
+        (signals "More than one value provided for parameter"))
+  (each (toplevel ('#$:m' 1 2))
+        (toplevel ('#$:m' ('#$:void') 2))       // Hidden: no value
+        (toplevel ('#$:m' ('#$:values' 1 2) 2)) // Hidden: more than one value
+        (signals "Too many arguments")))
+
+
+(ion_1_1 "? parameters"
+  (then (mactab
+          (macro m (v?) v))
+        (each (toplevel ('#$:m'))
+              (toplevel ('#$:m' ('#$:')))
+              (toplevel ('#$:m' ('#$:values')))
+              (produces))
+        (each (toplevel ('#$:m' 1))
+              (toplevel ('#$:m' ('#$:' 1)))   // TODO should this be allowed? No group in binary
+              (toplevel ('#$:m' ('#$:values' 1)))
+              (produces 1))
+        (each (toplevel ('#$:m' ('#$:values' 1 2)))
+              (signals "More than one value provided for parameter"))
+        (each (toplevel ('#$:m' 1 2))
+              (toplevel ('#$:m' 1 ('#$:values')))
+              (toplevel ('#$:m' ('#$:' 1 2) 3))
+              (toplevel ('#$:m' ('#$:values') 1))
+              (signals "Too many arguments")))
```

Comment:
I'd like to be extra clear about documenting the expected behavior here (and in the other test cases in this PR). I know that's asking for almost a full rewrite, so I can take it over if you prefer.
```suggestion
(ion_1_1 "a '?' parameter"
         (mactab (macro m (v?) v))
         (then "argument can be elided"
               (toplevel ('#$:m') (produces))
         (then "argument can be an empty expression group"
               (toplevel ('#$:m' ('#$:'))) (produces))
         (then "argument can be an e-expression that evaluates to void"
               (toplevel ('#$:m' ('#$:values'))) (produces))
         (then "can be a single expression"
               (toplevel ('#$:m' 1)) (produces 1))
         (then "can be an expression group with a single expression"
               // TODO should this be allowed? No group in binary
               (toplevel ('#$:m' ('#$:' 1))) (produces 1))
         (then "can be an e-expression that evaluates to a single expression"
               (toplevel ('#$:m' ('#$:values' 1))) (produces 1))
         (then "raises an error when there is more than one value produced by its argument's expression"
              (toplevel ('#$:m' ('#$:values' 1 2)))
              (signals "More than one value provided for parameter"))
         (each "raises an error when there are too many arguments"
              (toplevel ('#$:m' 1 2))
              (toplevel ('#$:m' 1 ('#$:values')))
              (toplevel ('#$:m' ('#$:' 1 2) 3))
              (toplevel ('#$:m' ('#$:values') 1))
              (signals "Too many arguments")))
```

Reply:
Yeah I think I'd prefer if you backfill more descriptions so we can get things exported asap

Reply:
Alright. I will take care of that.